app/assets/v2/js/cart.js

```diff
@@ -928,133 +928,207 @@ Vue.component('grants-cart', {
+      } catch (err) {
+        // Something went wrong, so we use the manual ingestion process instead
+        console.error(err);
+        console.log('Standard contribution ingestion failed, falling back to manual ingestion');
+        await this.postToDatabaseManualIngestion(txHash, userAddress);
+      }
+    },
+
+    // Alternative to postToDatabase that uses the manual ingestion process
+    async postToDatabaseManualIngestion(txHash, userAddress) {
+      // Determine if this was a zkSync checkout or standard L1 checkout. For this endpoint, we pass a txHash
+      // to ingest L1 contributions or an address to pass L2 contributions
+      const checkout_type = txHash[0].startsWith('sync') ? 'eth_zksync' : 'eth_std';
+      
+      txHash = checkout_type === 'eth_std' ? txHash[0] : ''; // txHash is always an array of hashes from checkout
+      userAddress = checkout_type === 'eth_zksync' ? userAddress : '';
+
+      // Get user's signature to prevent ingesting arbitrary transactions under your own username, then ingest
+      const { signature, message } = await this.signMessage(userAddress);
+      const csrfmiddlewaretoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
+      const url = '/grants/ingest';
+      const headers = { 'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8' };
+      const network = document.web3network || 'mainnet';
+      const payload = { csrfmiddlewaretoken, txHash, userAddress, signature, message, network };
+      const postParams = { method: 'POST', headers, body: new URLSearchParams(payload) };
+      let json; // response
 
       // Send saveSubscription request
-      const res = await fetch(url, saveSubscriptionParams);
-      const json = await res.json();
+      try {
+        _alert('Please be patient, as manual ingestion may take 1-2 minutes', 'info', '3000');
+        const res = await fetch(url, postParams);
+
+        json = await res.json();
+        console.log('ingestion response: ', json);
+        if (!json.success) {
+          console.log('ingestion failed');
+          throw new Error(`Your transactions could not be processed. Please visit ${window.location.host}/grants/add-missing-contributions to ensure your contributions are counted`);
+        }
+      } catch (err) {
+        console.error(err);
+        const message = `Your contribution was successful, but was not recognized by our database. Please visit ${window.location.host}/grants/add-missing-contributions to ensure your contributions are counted`;
+        
+        _alert(message, 'error');
+        throw new Error(message);
+      }
+    },
 
-      // if (json.failures.length > 0) {
-      //   // Something went wrong, so we create a backup of the users cart
-      //   await this.manageEthereumCartJSONStore(`${userAddress} - ${new Date().getTime()}`, 'save');
-      // }
+    // Asks user to sign a message as verification they own the provided address
+    async signMessage(userAddress) {
```

Comment:
can we DRY this with the func in the other file?

Reply:
Yes, but didn't for similar reason to here: https://github.com/gitcoinco/web/pull/8563#discussion_r593255675